src/main/java/org/reactome/release/Resource.java

```diff
@@ -0,0 +1,166 @@
+package org.reactome.release;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import com.google.gson.JsonParser;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.commons.csv.CSVRecord;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class Resource {
+	private final Logger logger = LogManager.getLogger();
+	private JsonObject resourceAsJson;
+
+	public Resource(CSVRecord resourceInfoCSV) {
+		this.resourceAsJson = convertCSVToJson(resourceInfoCSV);
+		checkMandatoryAttributesExist(this.resourceAsJson.toString());
+	}
+
+	public Resource(JsonObject resourceInfoJson) {
+		this.resourceAsJson = resourceInfoJson;
+		checkMandatoryAttributesExist(this.resourceAsJson.toString());
+	}
+
+	private JsonObject convertCSVToJson(CSVRecord resourceInfoCSV) {
+		String resourceInfoJson = convertCSVToJsonString(resourceInfoCSV);
+
+		return new JsonParser().parse(resourceInfoJson).getAsJsonObject();
+	}
+
+	private String convertCSVToJsonString(CSVRecord resourceInfoCSV) {
+		String resourceInfoJson = "";
+		try {
+			ObjectMapper mapper = new ObjectMapper();
+			mapper.enable(SerializationFeature.INDENT_OUTPUT);
+
+			resourceInfoJson = mapper.writeValueAsString(resourceInfoCSV.toMap());
+		} catch (JsonProcessingException e) {
+			logger.error("Unable to parse the CSV record's header to value map into JSON String", e);
+		}
+
+		return resourceInfoJson;
+	}
+
+	private void checkMandatoryAttributesExist(String originalRecord) throws IllegalArgumentException {
```

Comment:
I know you said this is a WIP so will probably plan to do this anyways, but for this method I'm a little unsure how it accomplishes what it does. I get the gist of it from the name of the method, but putting all these values into the Map and such is something I don't quite understand

Reply:
It is the how or the why that's unclear?

Reply:
The map holds the attributes of the resource that should be included in the JSON that describes the resource.  The key for the map is the name of the attribute that should be there and the value is the call to the appropriate "getter" method which should be able to provide that value since the record is now stored in the `Resource` object after being constructed.

If any of the values in the map give a null or empty value (checked for in the for loop iterating over the entries in the map), it means something is wrong because a value that should have been available after the construction of the `Resource` object is not available.  This triggers an `IllegalStateException` because the `Resource` object is not in the expected state of having values for fundamental information about the resource it represents.  This is either because the JSON provided didn't specify that information or something went wrong in providing that information during the construction of the object.  Either way, it's an illegal state and should cause an exception to check the problem with the resource info and/or code.